CONTRIBUTING.md

```diff
@@ -37,7 +37,33 @@ Please be sure to follow the usual process for submitting PRs:
 
 We reserve the right to close PRs that are not making progress. Closed PRs can be reopened again later and work can resume.
 
-### DCO Signing & Contributor License Agreement
+### Copyright Assignment
+
+GNU Radio does not claim ownership of any contributions you make. All copyrights remain with the original author of the contribution. As the author, you are responsible for maintaining and marking your copyright in the appropriate locations.
+
+When modifying or adding new files, you must include a copyright header in each file you touch. The copyright header should contain the following information:
+
+```
+/*
+ * Copyright (C) [Year] [Name or Pseudonym of Author]
+ *
+ * SPDX-License-Identifier: [License of module or file, default LGPL-3.0 for core]
```

Comment:
Since 3882ef71ac70f70a71aec455c6bd8b57e137b448, the SPDX for the core is `LGPL-3.0-linking-exception` rather than `LGPL-3.0`.

I'm just marking this as an item to review and keep in sync with the core `LICENSE` file before merging.

Reply:
We can safely revert this to the vanilla LGPL-3.0 w/o additional disclaimer. We got meanwhile the info that this has been explicitly [confirmed](http://listengine.tuxfamily.org/lists.tuxfamily.org/eigen/2009/01/msg00083.html) for another header-only C++ library (Eigen) by Brett Smith (Licensing Compliance Engineer, [Free Software Foundation](http://fsf.org/)).

For more practical details, see [here](https://www.eigen.tuxfamily.org/index.php?title=Licensing_FAQ&oldid=1117#So_what_does_the_LGPL_require_me_to_do.3F).

I hope this concludes this legal semantic debate.

Reply:
Is the link to the eigen mailing list correct? For me it goes to a thread "[eigen] problem to contact svn repository".

I'm not sure that eigen is a good example to advocate for LGPL-3.0 w/o linking exception for a header-only library. In 2012 [they changed their license to MPL 2.0](http://eigen.tuxfamily.org/index.php?title=ChangeLog#Eigen_3.1.1). In [their own words](http://eigen.tuxfamily.org/index.php?title=Licensing_FAQ), when eigen was LGPL they had a long FAQ about the implications that it isn't needed anymore with the MPL. It would be good to know more about the background and reasons for the license change.

The old [LGPL FAQ](http://eigen.tuxfamily.org/index.php?title=Licensing_FAQ&oldid=1117#So_what_does_the_LGPL_require_me_to_do.3F) mentions in two places that they asked the FSF about what happens with a header-only library, but the links they give to the mailing list are wrong (or most likely have become broken): http://listengine.tuxfamily.org/lists.tuxfamily.org/eigen/2009/01/msg00083.html and http://listengine.tuxfamily.org/lists.tuxfamily.org/eigen/2008/02/msg00008.html (one of these is the link you mentioned, the other points to a thread that does speak about FSF and licensing, but there are no clear answers there).

I think that in practice it doesn't matter if someone from the FSF in 2008 or 2009 said something about header-only libraries under the LGPL-3.0. What matters if we want to promote/simplify using GNU Radio 4.0 in closed source applications in industry is whether a legal department can review the license terms and have high confidence that there is no "GPL-like viral contamination" of the company's closed source code. If there is any reasonable doubt, they'll err on the side of caution and treat the license as a GPL-like viral license.

Reply:
@daniestevez corrected the link. We could also opt for a more liberal license, but the decision of whether LGPLv3, or more liberal one is not mine to make and should be moderated/coordinated by the GR leadership team.

The repo is technical, and we should find a different platform to discuss the nuances of GNU Radio Project leadership and roadmap issues.

Reply:
Thanks for finding the correct link. Still Brett Smith's response looks unconvincing to me. The question he is answering to is "I would like to know if the LGPLv3 can be suitable for a C++ pure template library, where all the code is in headers?", which is vague since "suitable" can mean any number of things. Brett simply says "yes, it is suitable, because of LGPLv3 Section 3".

The question that needs an answer is "can a closed-source application use an LGPLv3 header-only library, and if so, which requirements need to be followed". I know about section 3 in the LGPLv3, but it says "Give prominent notice with each copy of the object code that the Library is used in it and that the Library and its use are covered by this License". My reading is that Section 4 d) is still applicable, so users need to have a means to modify the Library in the Combined Work (and that's impossible if the Combined Work is a closed-source application and the Library is header-only).

> The repo is technical, and we should find a different platform to discuss the nuances of GNU Radio Project leadership and roadmap issues.

Happy to take the discussion somewhere else. I think the reason we're discussing in this PR is that this got brought up in the Architecture Matrix channel and between discussing it in the channel or discussing it in the PR, it was decided that it was better to discuss it in the PR.